### PR TITLE
Add automatic seed creation + automatic seeding

### PR DESCRIPTION
**Pull request by BenSeDev**
This PR will allow you to create a seed file in the same way as you create a migration file.

Similar to creating a new migration file:
vendor/bin/migraine new db_name "description which becomes the file name"

With this update you'll be able to run the following:
vendor/bin/migraine new-seed db_name "description which becomes the file name"

It will then create the seed file inside the appropriate seeds folder inside the folder named after the DB.

### CODE REVIEW

**Review by cedricengagor — COMMENTED** [review]
Wouldn't it make sense if we took the approach:
```sh
migraine new migration test "create user table"
migraine new seed test "default user"
```

This would basically follow the pattern:
`migraine {action=new} {type=seed} {group=test} {name="default user"}`

**Comment by BenSeDev** [comment]
> Wouldn't it make sense if we took the approach:
> 
> ```shell
> migraine new migration test "create user table"
> migraine new seed test "default user"
> ```
> 
> This would basically follow the pattern: `migraine {action=new} {type=seed} {group=test} {name="default user"}`

Done Ancient One.

**Review by matthias-deschoenmacker — APPROVED** [review]

File: src/NewSeed.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        $seedPath .= "/{$group}";
+
+        if (!is_dir($seedPath)) {
+            $this->filesystem->mkdir($seedPath);
+        }
+
+        $seedPath .= "/seeds";
+
+
+        if (!is_dir($seedPath)) {
```
Comment (cedricengagor) [inline src/NewSeed.php:78]:
```suggestion
        if (!is_dir($seedPath)) {
```

**Review by cedricengagor — APPROVED** [review]

**Comment by BenSeDev** [comment]
@matthias-deschoenmacker & @cedricengagor 

An addition to the PR: Allowing a `--seed` option to the migrate command.
Which will execute all the seed files & log them in the logs.json

File: src/SeedUp.php [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+    /**
+     * @param bool $commit
+     * @param string $onlyMigrateThisGroup
+     *
+     * @throws CouldNotGenerateConfig
+     *
+     * @return CompletedSeeds
+     */
+    public function seedUp(bool $commit = false, string $onlyMigrateThisGroup = ''): CompletedSeeds
```
Comment (cedricengagor) [inline src/SeedUp.php:40]: Unit test?

**Review by cedricengagor — CHANGES_REQUESTED** [review]
Nice work, perhaps a unit test?

File: src/CommandSeed.php [hunk 3]
```diff
@@ -70,7 +70,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $commit = (bool) $input->getOption('commit');
 
-        $completedMigrations = $this->migrateUp->seed($commit, $seed);
+        $completedMigrations = $this->seedUp->seed($commit, $seed);
 
         $listOfCompletedMigrations = $completedMigrations->getList();
```
[the remark this answers is not in the document]
Comment (BenSeDev) [reply src/CommandSeed.php]: Done

File: src/SeedUp.php (hunk 2 above)
Comment (BenSeDev) [reply src/SeedUp.php:40]: Sure thing Ancient One

File: src/LogsSQL.php [hunk 4]
```diff
@@ -63,6 +63,42 @@ public function migrationWasExecuted(LogStrategy $strategy, string $connectionSt
...
+
+        $events = $statement->fetchAll();
+
+        $events = $this->mapRowsToEvents($events);
+
+        foreach ($events as $event) {
+            $event = $event->toArray();
+
+            if (
+                in_array($event['event'], [EventSeedWasExecuted::class, EventMigrationWasSkipped::class])
```
Comment (cedricengagor) [inline src/LogsSQL.php:91]: Q: Do we need to check for migration was skipped here? I think this can 🔥 ?

File: src/LogsSQL.php [hunk 5]
```diff
@@ -63,6 +63,42 @@ public function migrationWasExecuted(LogStrategy $strategy, string $connectionSt
...
+        }
+
+        $db = new PDO(
+            $strategy->getConnectionString(),
+            $strategy->getUser(),
+            $strategy->getPassword(),
+            [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]
+        );
+
+        $query = "SELECT * FROM `{$strategy->getTable()}`";
```
Comment (cedricengagor) [inline src/LogsSQL.php:79]:
```suggestion
        $query = "SELECT `event` FROM `{$strategy->getTable()}`";
```

Looking from the code it seemed that only `event` was used. (but best to double check). It is always better to specify the required columns here, both for readability and to reduce bandwidth

**Review by cedricengagor — COMMENTED** [review]
1 more Q, but nice work. Looking forward to this 👏

File: src/LogsSQL.php (hunk 5 above)
Comment (BenSeDev) [reply src/LogsSQL.php]: So this was a copy pasta, but indeed, for the seeds we're only checking for the event, so, you ask wij draaien eh.

File: src/LogsSQL.php (hunk 4 above)
Comment (BenSeDev) [reply src/LogsSQL.php]: You've passed the test Ancient One. 
Eternal fame for noticing, 🤣

**Review by cedricengagor — APPROVED** [review]

**Review by matthias-deschoenmacker — APPROVED** [review]
LGTM

**Pull request merged**